### PR TITLE
[doc] Various documentation fixes

### LINKED ISSUE #1856: Example in `find_modules()` documentation is wrong
This code listing in the documentation [here](http://localhost:8000/utility_functions_reference.html#reframe.utility.find_modules) is wrong:

```python
my_find_modules = functools.partial(find_modules, environ_mapping={
    r'.*CrayGNU.*': {'PrgEnv-gnu'},
    r'.*CrayIntel.*': {'PrgEnv-intel'},
    r'.*CrayCCE.*': {'PrgEnv-cray'}
})
```

It should be

```python
my_find_modules = functools.partial(find_modules, environ_mapping={
    r'.*CrayGNU.*': 'PrgEnv-gnu',
    r'.*CrayIntel.*': 'PrgEnv-intel',
    r'.*CrayCCE.*': 'PrgEnv-cray'
})
```

Otherwise you get a `TypeError`.

### LINKED ISSUE #1858: Deprecation message is missing from DEPEND_* constants in the documentation

### PR DESCRIPTION
**Pull request by vkarak**
Fixes #1856.
Fixes #1858.

### CODE REVIEW

File: reframe/utility/__init__.py [hunk 1]
```diff
@@ -565,10 +565,12 @@ def find_modules(substr, environ_mapping=None):
 
     .. code:: python
 
-       @rfm.parameterized_test(*find_modules('netcdf'))
+       @rfm.simple_test
        class MyTest(rfm.RegressionTest):
-           def __init__(self, s, e, m):
-               self.descr = f'{s}, {e}, {m}'
```
Comment (ekouts) [inline reframe/utility/__init__.py:571]: I am not sure if you remove the description by mistake here?

File: reframe/utility/__init__.py [hunk 2]
```diff
@@ -583,15 +585,17 @@ def __init__(self, s, e, m):
...
+           r'.*CrayGNU.*': 'PrgEnv-gnu',
+           r'.*CrayIntel.*': 'PrgEnv-intel',
+           r'.*CrayCCE.*': 'PrgEnv-cray'
        })
 
-       @rfm.parameterized_test(*my_find_modules('GROMACS'))
+       @rfm.simple_test
        class MyTest(rfm.RegressionTest):
-           def __init__(self, s, e, m):
-               self.descr = f'{s}, {e}, {m}'
```
Comment (ekouts) [inline reframe/utility/__init__.py:594]: Same here.

**Review by ekouts — APPROVED** [review]
Lgtm overall, I am not sure if you omit the description on purpose but it can be merged as it is.

File: reframe/utility/__init__.py (hunk 1 above)
Comment (vkarak) [reply reframe/utility/__init__.py:571]: I did it no purpose. It's not needed for what we need to show.

**Pull request merged**